src/Datadog.Trace/ExtensionMethods/StringExtensions.cs

```diff
@@ -69,5 +70,65 @@ public static string TrimEnd(this string value, string suffix, StringComparison
 
             return null;
         }
+
+        /// <summary>
+        /// Datadog tag requirements:
+        /// 1. Tag must start with a letter
+        /// 2. Tag cannot exceed 200 characters
+        /// 3. If the first two requirements are met, then valid characters will be retained while all other characters will be converted to underscores. Valid characters include:
+        ///    - Alphanumerics
+        ///    - Underscores
+        ///    - Minuses
+        ///    - Colons
+        ///    - Periods
+        ///    - Slashes
+        ///
+        /// Note: This method will trim leading/trailing whitespace before checking the requirements.
+        /// </summary>
+        /// <param name="value">Input string to convert into tag name</param>
+        /// <param name="normalizedTagName">If the method returns true, the normalized tag name</param>
+        /// <param name="convertPeriodsToUnderscores">A flag determining whether periods should also be converted to underscores</param>
+        /// <returns>Returns whether the conversion was successful</returns>
+        public static bool TryConvertToNormalizedTagName(this string value, out string normalizedTagName, bool convertPeriodsToUnderscores = false)
```

Comment:
Since this method is internal, I would make `convertPeriodsToUnderscores` non-optional to force us to be explicit about it in every call. To me the optional parameter is a convenience for users in public APIs. (level of care: low)

Reply:
That's a good suggestion, will do

Reply:
Done in c1ac5ce1868b7c359e741bd04d05e742647a9a33

Reply:
On second thought (sorry!), do we need the overload without `convertPeriodsToUnderscores` at all?

Reply:
We don't technically need it in this set of changes...but we kind of should be normalizing the tag names that users give us in `SetTag` 😄

Reply:
My mistake! I thought they were overloads (same name), but it's `TryConvertToNormalizedTagName` vs `TryConvertToNormalizedHeaderTagName`. Resolving this comment.